### PR TITLE
Fix memory leak in parquet reader gzip decompression

### PR DESCRIPTION
**Pull request by Yuhta**
Differential Revision: D38874446

### CODE REVIEW

**Review by majetideepak — APPROVED** [review]

**Comment by majetideepak** [comment]
@Yuhta thanks for fixing this.

File: velox/dwio/parquet/reader/PageReader.cpp [hunk 1]
```diff
@@ -188,6 +188,20 @@ const char* FOLLY_NONNULL PageReader::uncompressData(
...
+        explicit InflateEndGuard(z_stream* stream) : stream_(stream) {}
+        ~InflateEndGuard() {
+          if (inflateEnd(stream_) != Z_OK) {
+            LOG(WARNING) << "inflateEnd: "
+                         << (stream_->msg ? stream_->msg : "");
+          }
+        }
+
+       private:
+        z_stream* stream_;
```
Comment (xiaoxmeng) [inline velox/dwio/parquet/reader/PageReader.cpp:202]:
```
nit: z_stream* const stream_;
```

File: velox/dwio/parquet/reader/PageReader.cpp [hunk 2]
```diff
@@ -188,6 +188,20 @@ const char* FOLLY_NONNULL PageReader::uncompressData(
           (ret == Z_OK),
           "zlib inflateInit failed: {}",
           stream.msg ? stream.msg : "");
+
+      struct InflateEndGuard {
+        explicit InflateEndGuard(z_stream* stream) : stream_(stream) {}
```
Comment (xiaoxmeng) [inline velox/dwio/parquet/reader/PageReader.cpp:193]:
```
VELOX_CHECK_NOTNULL(stream_)
```

**Review by xiaoxmeng — APPROVED** [review]
Thanks for the fix!

File: velox/dwio/parquet/reader/PageReader.cpp (hunk 1 above)
Comment (Yuhta) [reply velox/dwio/parquet/reader/PageReader.cpp:202]: `inflateEnd` takes a non-const pointer

File: velox/dwio/parquet/reader/PageReader.cpp (hunk 2 above)
Comment (Yuhta) [reply velox/dwio/parquet/reader/PageReader.cpp:193]: This is just a local struct and we are guaranteed to pass a non-null pointer

**Pull request closed without merging**